data_creation/data_creation.py

```diff
@@ -0,0 +1,357 @@
+import os
+import importlib
+
+import numpy as np
+import tensorflow as tf
+import tensorflow_datasets as tfds
+from functools import lru_cache
+import math
+
+from sign_language_datasets.datasets.config import SignDatasetConfig
+from sign_language_datasets.datasets.dgs_corpus import DgsCorpusConfig
+import sign_language_datasets.datasets
+
+from pose_format import PoseHeader, Pose
+from pose_format.numpy.pose_body import NumPyPoseBody
+from pose_format.tensorflow.pose_body import TensorflowPoseBody
+from pose_format.utils.reader import BufferReader
+from spoken_to_signed.gloss_to_pose import concatenate_poses
+
+# Project-Specific Imports
+from dgs_types_mapping import create_gloss_to_pose_dict
+
+
+
+
+class DGSPoseDataset:
+    def __init__(self, corpus_dir, dictionary_dir, num_examples=None):
+        """
+        Initializes the DGSPoseDataset class.
+        
+        :param corpus_dir: Directory to load DGS Corpus.
+        :param dictionary_dir: Directory to load DGS Types.
+        :param num_examples: Maximum number of examples to process (default: None, process all examples).
```

Comment:
if `num_examples` is a maximum, it should be `max_examples` or something similar. the code is more important than the doc